### PR TITLE
Feat: Conclusion redesign

### PR DESCRIPTION
**Pull request by svenjens**
Description 
This PR Takes care of small design changes in Conclusion
Please Squash & Merge 

Please make sure:
- [x] to add a label
- [x] to add one or more reviewers
- [x] you followed our checklist for [functional testing](https://github.com/Amsterdam/vergunningcheck/blob/master/TESTING.md)
- [x] you added the necessary documentation (either in the markdown files or inline)
- [x] you added the necessary automated tests

### CODE REVIEW

File: apps/client/src/atoms/ComponentWrapper.ts [hunk 1]
```diff
@@ -1,7 +1,11 @@
 import { themeSpacing } from "@datapunt/asc-ui";
 import styled, { css } from "styled-components";
 
-export default styled.div`
+type Props = {
```
Comment (robinpiets) [inline apps/client/src/atoms/ComponentWrapper.ts:4]: 👍

File: apps/client/src/atoms/PrintButton.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  ${(props) =>
+    css`
+      margin-top: ${props.marginTop ? props.marginTop + "px" : themeSpacing(5)};
+      margin-bottom: ${props.marginBottom
+        ? props.marginBottom + "px"
+        : themeSpacing(5)};
+    `}
+  @media print {
+    display: none;
+  }
```
Comment (robinpiets) [inline apps/client/src/atoms/PrintButton.ts]: This is not necessary because all PrintButtons are called within `<HideForPrint />` components

File: apps/client/src/components/Conclusion.js [hunk 3]
```diff
@@ -78,42 +69,24 @@ const Conclusion = ({ checker, matomoTrackEvent }) => {
...
-              onClick={handlePrintButton}
-              type="button"
-            >
-              Conclusie opslaan
-            </PrintButton>
-          </ComponentWrapper>
-        )}
-      </HideForPrint>
+      {needsPermit && !contactConclusion && (
+        <NeedPermit
```
Comment (robinpiets) [inline apps/client/src/components/Conclusion.js]: I find these three components to have inconsistent names: NeedPermit, PermitFree, ContactConclusion

Also, the content of these files is pretty similar and the props you're passing is similar. They all have this flow:
- Title
- Optional text (Markdown or Paragraph)
- Buttons
- Optional text

I think it would be better to make one variable called `conclusionType` which can only be 'needPermit', 'needContact' and 'permitFree'. For each of these `conclusionType` make a configurable object with a title, optional texts (Components) and buttons. Then you can build the content of this page dynamically instead of having a lot of duplicate code.

What do you guys think? @afjlambert @svenjens

**Review by robinpiets — CHANGES_REQUESTED** [review]

**Review by afjlambert — APPROVED** [review]

File: apps/client/src/components/Conclusion.js [hunk 4]
```diff
@@ -1,20 +1,15 @@
...
-} from "../atoms";
+import { Alert, PrintOnly } from "../atoms";
 import { Olo } from "../config";
-import { actions, eventNames, sections } from "../config/matomo";
+import { actions, eventNames } from "../config/matomo";
 import withTracking from "../hoc/withTracking";
 import { sttrOutcomes } from "../sttr_client/models/checker";
+import { ContactConclusion } from "./Conclusion/ContactConclusion";
+import { NeedPermit } from "./Conclusion/NeedPermit";
+import { PermitFree } from "./Conclusion/PermitFree";
```
Comment (afjlambert) [inline apps/client/src/components/Conclusion.js]: Maybe add an index.ts to that folder so you can easily import them all?

File: apps/client/src/components/Conclusion/ContactConclusion.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import { sections } from "../../config/matomo";
+import Markdown from "../Markdown";
+
+type Props = {
+  contactConclusion: { [key: string]: string };
+  handlePrintButton: () => void;
+};
+
+export const ContactConclusion: React.FC<
+  Props & React.HTMLAttributes<HTMLElement>
```
Comment (afjlambert) [inline apps/client/src/components/Conclusion/ContactConclusion.tsx]: Can you remove `React.HTMLAttributes<HTMLElement>` ?

File: apps/client/src/components/Conclusion/ContactConclusion.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import { Heading } from "@datapunt/asc-ui";
+import React from "react";
+import { isIE, isMobile } from "react-device-detect";
+
+import { HideForPrint, PrintButton } from "../../atoms";
+import { sections } from "../../config/matomo";
+import Markdown from "../Markdown";
+
+type Props = {
+  contactConclusion: { [key: string]: string };
```
Comment (afjlambert) [inline apps/client/src/components/Conclusion/ContactConclusion.tsx]: Why did you go for `{ [key: string]: string }`?
Maybe specify `title` and `description` here?

File: apps/client/src/components/Conclusion.js [hunk 7]
```diff
@@ -78,42 +69,24 @@ const Conclusion = ({ checker, matomoTrackEvent }) => {
...
-            >
-              Conclusie opslaan
-            </PrintButton>
-          </ComponentWrapper>
-        )}
-      </HideForPrint>
+      {contactConclusion && (
+        <ContactConclusion
+          contactConclusion={contactConclusion}
+          handlePrintButton={handlePrintButton}
```
Comment (afjlambert) [inline apps/client/src/components/Conclusion.js]: Would be nice if we could solve `handlePrintButton` in a different way so we don't have to drill props here.

File: apps/client/src/components/Conclusion/NeedPermit.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+import Link from "../Link";
+
+type Props = {
+  contactConclusion: object;
+  handlePermitButton: () => void;
+  handlePrintButton: () => void;
+};
+
+export const NeedPermit: React.FC<
+  Props & React.HTMLAttributes<HTMLElement>
```
Comment (afjlambert) [inline apps/client/src/components/Conclusion/NeedPermit.tsx]: Please remove `React.HTMLAttributes<HTMLElement>`

File: apps/client/src/components/Conclusion/NeedPermit.tsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+          Conclusie opslaan
+        </PrintButton>
+      )}
+    </HideForPrint>
+
+    <Heading forwardedAs="h2">Meer weten?</Heading>
+    <Paragraph>
+      Wilt u weten hoe de aanvraag werkt, wat de kosten zijn of waar u nog meer
+      aan moet denken als u gaat starten? Op onze pagina
+      <Link>omgevingsvergunning</Link> is alle informatie te vinden.
```
Comment (afjlambert) [inline apps/client/src/components/Conclusion/NeedPermit.tsx]: `<Link>` without href?

File: apps/client/src/components/Conclusion/PermitFree.tsx [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+import { List } from "../../atoms";
+import { HideForPrint, PrintButton } from "../../atoms";
+
+type Props = {
+  contactConclusion: object;
+  handlePrintButton: () => void;
+};
+
+export const PermitFree: React.FC<
+  Props & React.HTMLAttributes<HTMLElement>
```
Comment (afjlambert) [inline apps/client/src/components/Conclusion/PermitFree.tsx]: same here.

File: apps/client/src/components/Conclusion/PermitFree.tsx [hunk 11]
```diff
@@ -0,0 +1,58 @@
+import { Heading, ListItem } from "@datapunt/asc-ui";
+import React from "react";
+import { isIE, isMobile } from "react-device-detect";
+
+import { List } from "../../atoms";
+import { HideForPrint, PrintButton } from "../../atoms";
+
+type Props = {
+  contactConclusion: object;
```
Comment (afjlambert) [inline apps/client/src/components/Conclusion/PermitFree.tsx]: This should not be a contactConclusion.

File: apps/client/src/components/Conclusion/PermitFree.tsx [hunk 12]
```diff
@@ -0,0 +1,58 @@
+import { Heading, ListItem } from "@datapunt/asc-ui";
+import React from "react";
+import { isIE, isMobile } from "react-device-detect";
+
+import { List } from "../../atoms";
+import { HideForPrint, PrintButton } from "../../atoms";
```
Comment (afjlambert) [inline apps/client/src/components/Conclusion/PermitFree.tsx]: Please merge imports.

**Review by afjlambert — CHANGES_REQUESTED** [review]

File: apps/client/src/components/Conclusion.js (hunk 3 above)
Comment (afjlambert) [reply apps/client/src/components/Conclusion.js]: Yeah I agree. Something similar to https://github.com/Amsterdam/vergunningcheck/pull/475/files maybe?

File: apps/client/src/components/ConclusionOutcome.tsx [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+      )}
+      {!needsPermit && !contactConclusion && (
+        <>
+          <Paragraph>
+            U hebt geen vergunning nodig voor {topic?.name.toLowerCase()}. Wél
+            moet u op een aantal dingen letten voordat u gaat beginnen. Uw
+            aannemer kan u daarbij helpen.
+          </Paragraph>
+          <Heading forwardedAs="h3">
+            Dit is waar u verder op moet letten:
```
Comment (robinpiets) [inline apps/client/src/components/ConclusionOutcome.tsx]: Let's put this in a separate component

File: apps/client/src/components/ConclusionOutcome.tsx [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+              variant={"inline"}
+              href={
+                "https://www.amsterdam.nl/veelgevraagd/?productid=%7B215DE049-EFA3-492D-A4B1-EDFF40E0BC51%7D#case_%7BC7581CB6-A3F4-4FCA-B631-C93EF03A108D%7D"
+              }
+            >
+              omgevingsvergunning
+            </Link>{" "}
+            is alle informatie te vinden.
+          </Paragraph>
+        </>
```
Comment (robinpiets) [inline apps/client/src/components/ConclusionOutcome.tsx]: Let's put this in a separate component

File: apps/client/src/components/Conclusion.js [hunk 15]
```diff
@@ -53,67 +45,17 @@ const Conclusion = ({ checker, matomoTrackEvent }) => {
...
-    });
-    window.print();
-  };
-
   return (
     <>
-      <Paragraph>
-        Op basis van uw antwoorden vindt u hieronder wat voor uw activiteit van
-        toepassing is.
+      <Paragraph gutterBottom={isMobile ? 16 : 20}>
```
Comment (robinpiets) [inline apps/client/src/components/Conclusion.js]: Can you add this to the ConclusionOutcome as well?

File: apps/client/src/components/ConclusionOutcome.tsx [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+  contactConclusion: { title: string; description: string };
+  needsPermit: boolean;
+  matomoTrackEvent: Function;
+};
+
+export const ConclusionOutcome: React.FC<Props> = ({
+  contactConclusion,
+  needsPermit,
+  matomoTrackEvent,
+}) => {
```
Comment (robinpiets) [inline apps/client/src/components/ConclusionOutcome.tsx]: NP: abc

File: apps/client/src/components/ConclusionOutcome.tsx [hunk 17]
```diff
@@ -0,0 +1,147 @@
...
+      {needsPermit && (
+        <>
+          <Heading forwardedAs="h2">Meer weten?</Heading>
+          <Paragraph>
+            Wilt u weten hoe de aanvraag werkt, wat de kosten zijn of waar u nog
+            meer aan moet denken als u gaat starten? Op onze pagina
+            <Link
+              variant={"inline"}
+              href={
+                "https://www.amsterdam.nl/veelgevraagd/?productid=%7B215DE049-EFA3-492D-A4B1-EDFF40E0BC51%7D#case_%7BC7581CB6-A3F4-4FCA-B631-C93EF03A108D%7D"
```
Comment (robinpiets) [inline apps/client/src/components/ConclusionOutcome.tsx]: Can you remove the hash `#case_%7BC7581CB6-A3F4-4FCA-B631-C93EF03A108D%7D` from the link?

**Review by robinpiets — CHANGES_REQUESTED** [review]

File: apps/client/src/components/ConclusionOutcome.tsx [hunk 18]
```diff
@@ -0,0 +1,147 @@
+import { Button, Heading, ListItem, Paragraph } from "@datapunt/asc-ui";
```
Comment (robinpiets) [inline apps/client/src/components/ConclusionOutcome.tsx]: Can you move this file to components/Conclusion/ConclusionOutcome.tsx ? So we can add all conclusion related files to this.

File: apps/client/src/components/ConclusionOutcome.tsx [hunk 19]
```diff
@@ -0,0 +1,147 @@
...
+      )}
+      <HideForPrint>
+        {needsPermit && (
+          <ComponentWrapper marginBottom={32}>
+            <Button
+              color="secondary"
+              onClick={handlePermitButton}
+              type="button"
+            >
+              Vergunning aanvragen
```
Comment (robinpiets) [inline apps/client/src/components/ConclusionOutcome.tsx]: Can you add a small test to make sure "Vergunning aanvragen" is not rendered if conclusion is not `needsPermit`? So we don't have issues with that anymore. You can do this on a separate PR, so we do the acceptance test in the meantime.

File: apps/client/src/components/ConclusionOutcome.tsx (hunk 17 above)
Comment (svenjens) [reply apps/client/src/components/ConclusionOutcome.tsx]: We need the hash to open the right section

**Review by robinpiets — APPROVED** [review]
Please Squash and merge
